### DISCUSSION TITLE
activeButtonsInReadOnly: ['source', 'preview'] is not working.

### DISCUSSION THREAD

**Discussion by patiencendiligence** [xdan/jodit#877] [Q&A] [answered]
Hello, Somebody help me. plz.
(My jodit is the latest version. - free)

I want to the buttons such as 'source', 'preview' are working in readonly state.
So, I set like this my jodit editor config.
```
config: {
  readonly: true,
  activeButtonsInReadOnly: ['source', 'preview'],
  ...,
}
```

Then, The 'source' button works.
But.. The preview button looks like working - is active - . Actually, It is not working.
When I click the preview button, it is nothing happens. (not visible preview dialog..)

How can I make that preview button work in readonly state?
Thank you in advance.

**Comment by xdan** [comment]
`source` works because it only toggles the source view, but `preview`'s action looks like it's gated behind the readonly check — the button shows active but the command no-ops. That's arguably a bug: `activeButtonsInReadOnly` should let the listed buttons actually run.

Does it still reproduce on the current version? If so it's worth tracking as an issue so the preview command is allowed for readonly-active buttons. As a stopgap you can wire a small custom button that opens your own preview (drop `editor.value` into a dialog or new window).

**Comment by xdan** [comment, marked as answer]
This works on current versions. `preview` runs in readonly because it's in the default `allowCommandsInReadOnly` (`['selectall', 'preview', 'print']`) — that option was added in the fix for #878, after you reported this. So `readonly: true` together with `activeButtonsInReadOnly: ['source', 'preview']` opens the preview dialog as expected.

I just checked your exact config on the current build and the preview dialog opens with the content, so if it no-ops for you you're on an older build from before that option existed — upgrade to the latest and it'll work.
